r/tests/testthat.R

```diff
@@ -19,4 +19,4 @@ library(testthat)
 library(arrow)
 library(tibble)
 
-test_check("arrow")
+test_check("arrow", reporter="location")
```

Comment:
We will probably want to make this change only happen on CI so that the standard reporter is used when we submit to CRAN. We could have this trigger on `ARROW_R_DEV` which is (almost?) always set on CI and isn't a terrible overloading of the meaning. 

We also could use `reporter = MultiReporter("check", "location")` which should give us decent info in the check log, but also more detailed information for the artifact (this config basically concatenates the two reporter outputs in the output, but in reverse order)

Reply:
The environment variable check could be something like:

```
if (identical(tolower(Sys.getenv("ARROW_R_DEV", FALSE)), "true")) {
  arrow_reporter <- MultiReporter("check", "location")
} else {
  arrow_reporter <- check_reporter()
}
test_check("arrow", reporter = arrow_reporter)
```

Reply:
Done, thanks!